### ISSUE TITLE
Unstable integration test aws_ses_identity

### ISSUE THREAD

**Issue by mattclay** [ansible/ansible#36065]
##### ISSUE TYPE

Bug Report

##### COMPONENT NAME

aws_ses_identity integration test

##### ANSIBLE VERSION

2.5

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

Shippable

##### SUMMARY

Unstable integration test aws_ses_identity:

https://app.shippable.com/github/ansible/ansible/runs/54391/66/tests

##### STEPS TO REPRODUCE

Run tests on Shippable.

##### EXPECTED RESULTS

Tests pass.

##### ACTUAL RESULTS

Tests are unstable or fail.

> test/integration/targets/aws_ses_identity/tasks/main.yaml:242 / [localhost] amazon: aws_ses_identity : register email identity aws_secret_key={{ ec2_secret_key }}, aws_access_key={{ ec2_access_key }}, security_token={{security_token}}, region={{ ec2_region }}, state=present, identity={{ email_identity }}

> error: TypeError: argument of type 'NoneType' is not iterable

```
Traceback (most recent call last):
  File "/tmp/ansible_FTBGdl/ansible_module_aws_ses_identity.py", line 448, in <module>
    main()
  File "/tmp/ansible_FTBGdl/ansible_module_aws_ses_identity.py", line 443, in main
    create_or_update_identity(connection, module, region, account_id)
  File "/tmp/ansible_FTBGdl/ansible_module_aws_ses_identity.py", line 372, in create_or_update_identity
    notifications_changed, notification_attributes = update_identity_notifications(connection, module)
  File "/tmp/ansible_FTBGdl/ansible_module_aws_ses_identity.py", line 347, in update_identity_notifications
    changed |= update_notification_topic(connection, module, identity, identity_notifications, notification_type)
  File "/tmp/ansible_FTBGdl/ansible_module_aws_ses_identity.py", line 275, in update_notification_topic
    if topic_key in identity_notifications:
TypeError: argument of type 'NoneType' is not iterable
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/amazon/aws_ses_identity.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/aws_ses_identity.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @orthanc @willthames
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by orthanc** [comment]
Slightly odd, it looks like what is happening is that the boto call for notification settings isn't returning the information for the identity that's just been created so it's falling into an error case I wasn't expecting.

I'm guessing that this is becaue there's a delay on the AWS side between creation of the identity and the initial notification settings being avaliable through this API.

I'll have a look at the weekend and see if I can replicate this locally, pretty sure this can be addressed either by just returning the default setup when there are no results from the API, or possibly just looping with a slight delay to deal with eventual consistency in the API.

**Comment by orthanc** [comment]
Linked PR addresses the possible causes of instability I've been able to find, there are three I've run into:

1. Sometimes the API call to get notifications doesn't return information for the registered identity resulting in the error reported in the issue description.
1. The shippable build running python 2 and python 3 tests concurrently can exceed the max allowed calls to the SES APIs, resulting in throtteling errors
1. Sometimes the call to get the identity verification status doesn't return information for the registered identity resulting in an error serializing the successful response since we try to convert None to a snake case dict

I've been able to reproduce 2 by running the test concurrently on EC2 instances. I've had no such luck with 1 & 3, I've only observed these with shippable builds, so I'm only guessing at the cause. I'm fairly sure these are eventual consistency issues with the AWS apis, so I've put in retry handling for 3 and made the module tolerate against 1.

2 was addressed by increasing the number of retries in boto as that seems to be intended to deal with throttling.

_Same message is added to the PR_
